File: factory/code-typescript/premiereMethode/ProduitFactory.ts
import { Produit } from "./Produit";
import { Produit1 } from "./produitConcrets/Produit1.js";
import { Produit2 } from "./produitConcrets/Produit2.js";

export class ProduitFactory {
  public static TYPE_PRODUIT1: number = 1;
  public static TYPE_PRODUIT2: number = 2;

  constructor() {}

  getProduit(typeProduit: number): Produit {
    let produit: Produit;
    switch (typeProduit) {
      case ProduitFactory.TYPE_PRODUIT1:
        produit = new Produit1();
        break;
      case ProduitFactory.TYPE_PRODUIT2:
        produit = new Produit2();
        break;
      default:
        throw Error("Ce type de produit n'existe pas !");
    }

    return produit;
  }
}

File: 1-patrons-de-creations/factory/code-java/premiereMethode/produit_factory/ProduitFactory.java
package produit_factory;

import produits_concrets.Produit1;
import produits_concrets.Produit2;
import produit.Produit;

public class ProduitFactory {

    public static final int TYPE_PRODUIT1 = 1;
    public static final int TYPE_PRODUIT2 = 2;

    // La factory method
    public Produit getProduit(int typeProduit) {
        Produit produit = null;
        switch(typeProduit){
            case TYPE_PRODUIT1:
                produit = new Produit1();
                break;
            case TYPE_PRODUIT2:
                produit = new Produit2();
                break;
            default:
                throw new IllegalArgumentException("Type de produit inconnu");
        }

        return produit;
    }
}